src/algebra/finite_map/mod.rs

```diff
@@ -189,6 +189,28 @@ pub struct FiniteHashMap<Key, Value> {
 
 shared_ref_self_generic!(<Key, Value>, FiniteHashMap<Key, Value>);
 
+impl<Key, Value> NumEntries for FiniteHashMap<Key, Value>
+where
+    Key: KeyProperties,
+    Value: GroupValue + NumEntries,
+{
+    fn num_entries(&self) -> usize {
+        match Value::const_num_entries() {
+            None => {
+                let mut res = 0;
+                for (_, v) in self.into_iter() {
+                    res += v.num_entries();
+                }
+                res
+            }
```

Comment:
```suggestion
            None => self.into_iter().map(NumEntries::num_entries).sum(),
```
Also, couldn't/shouldn't that just be an `.iter()` call instead of an `.into_iter()` one?